buffer/src/main/java/io/netty/buffer/CompositeByteBuf.java

```diff
@@ -1111,26 +1256,35 @@ public int nioBufferCount() {
 
     @Override
     public ByteBuffer internalNioBuffer(int index, int length) {
-        if (components.size() == 1) {
+        switch (components.size()) {
+        case 0:
+            return EMPTY_NIO_BUFFER;
+        case 1:
             return components.get(0).buf.internalNioBuffer(index, length);
+        default:
+            throw new UnsupportedOperationException();
         }
-        throw new UnsupportedOperationException();
     }
 
     @Override
     public ByteBuffer nioBuffer(int index, int length) {
-        if (components.size() == 1) {
+        checkIndex(index, length);
+
+        switch (components.size()) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add a default case to this switch. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3ASwitchLastCaseIsDefaultCheck)